src/applications/mhv-medical-records/components/MrBreadcrumbs.jsx

```diff
@@ -1,45 +1,63 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
-import { Link, useHistory } from 'react-router-dom';
+import { VaBreadcrumbs } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import React, { useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Breadcrumbs, Paths } from '../util/constants';
+import { setBreadcrumbs } from '../actions/breadcrumbs';
 
 const MrBreadcrumbs = () => {
-  const crumbs = useSelector(state => state.mr.breadcrumbs.list);
-  const isDetails = useSelector(state => state.mr.isDetails.currentIsDetails);
-  const crumb = crumbs[0]?.url?.toLowerCase();
+  const dispatch = useDispatch();
+  const location = useLocation();
   const history = useHistory();
+  const crumbsList = useSelector(state => state.mr.breadcrumbs.crumbsList);
+
+  const [locationBasePath, locationChildPath] = useMemo(
+    () => {
+      const pathElements = location.pathname.split('/');
+      if (pathElements[0] === '') pathElements.shift();
+      return pathElements;
+    },
+    [location],
+  );
+
+  const headingText = document.querySelector('h1');
+
+  useEffect(
+    () => {
+      const path = locationBasePath ? `/${locationBasePath}/` : '/';
+      const feature = Object.keys(Paths).find(_path => path === Paths[_path]);
+
+      if (path === '/') {
+        dispatch(setBreadcrumbs([]));
+      } else if (locationChildPath && headingText) {
+        const detailCrumb = {
+          href: `${path}${locationChildPath}`,
+          label: headingText.textContent,
+          isRouterLink: true,
+        };
+        dispatch(setBreadcrumbs([Breadcrumbs[feature], detailCrumb]));
+      } else {
+        dispatch(setBreadcrumbs([Breadcrumbs[feature]]));
+      }
+    },
+    [dispatch, locationBasePath, locationChildPath, headingText],

```

Comment:
You should be able to shrink the scope of the dependency array a bit by using `headingText?.textContent` instead of `headingText`.

Reply:
Good point!